### PR TITLE
rename module, fix versioning

### PR DESCRIPTION
**Pull request by hairyhenderson**
Fixing versioning and renaming the Go module.

The version is now generated based on git tags. This also updates the metrics to expose an `smtprelay_build_info` metric.

### CODE REVIEW

File: scripts/version.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		return fmt.Errorf("failed to run cmd %s: %w", cmd, err)
+	}
+
+	fmt.Printf("v%s is tagged and pushed!\n", nextVer)
+
+	ghReleaseCmd := fmt.Sprintf("gh release create v%s --generate-notes", nextVer)
+	out, err = runCmd(ghReleaseCmd)
+	fmt.Println(out)
+	if err != nil {
+		if strings.Contains(err.Error(), `"gh": executable file not found in $PATH`) {
```
Comment (owensmallwood) [inline scripts/version.go:184]: Could we check `gh` is installed at the beginning of this function?

Comment (hairyhenderson) [reply scripts/version.go:184]: probably - this was copy/pasted from elsewhere though so I should probably do that in the source first

File: README.md [hunk 2]
```diff
@@ -33,11 +30,10 @@ produces mail.
...
 - `make build` - build go code
 - `make docker` build docker image
-- `make docker-push` - build and push docker image
-
-Update `DOCKER_TAG` in `config.mk` to change Docker Repo.
+- `make docker-tag` - build and push docker image
+- `make docker-push` - build, tag, and push docker image
 
-note: tags for docker images are generated by `scripts/version`
+Update `DOCKER_TAG` in `Makefile` to change Docker Repo.
```
Comment (owensmallwood) [inline README.md:36]: I think this should be be `DOCKER_IMAGE`

File: scripts/version.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+	switch {
+	case *next:
+		nextVer := ver.IncPatch()
+		fmt.Println((&nextVer).String())
+	case *releaseTag:
+		err := release(ver, *nosign)
+		logAndExit(err)
+	case !*goflagsVar:
+		fmt.Printf("%s\n", ver)
+	default:
```
Comment (owensmallwood) [inline scripts/version.go:48]: When/why do we call this with the `goFlagsVar` flag?

Comment (hairyhenderson) [reply scripts/version.go:48]: here: https://github.com/grafana/smtprelay/pull/45/files/6a21a6fc71409774814c96fe1b9dd898734b9ef7#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R7

**Review by owensmallwood — APPROVED** [review]
LGTM 🚀

**Pull request merged**